pkg/domain/runaway.go

```diff
@@ -308,7 +306,28 @@ func (do *Domain) handleRunawayWatch(record *resourcegroup.QuarantineRecord) err
 	}()
 	sql, params := record.GenInsertionStmt()
 	_, err = exec.ExecuteInternal(ctx, sql, params...)
-	return err
+	if err != nil {
+		return 0, err
+	}
+	for retry := 1; retry <= maxIDRetries; retry++ {
+		var rs sqlexec.RecordSet
+		rs, err = exec.ExecuteInternal(ctx, `SELECT LAST_INSERT_ID();`)
+		if err == nil {
+			var rows []chunk.Row
+			rows, err = sqlexec.DrainRecordSet(ctx, rs, 1)
+			//nolint: errcheck
+			rs.Close()
+			if err == nil {
+				if len(rows) == 1 {
+					return rows[0].GetUint64(0), nil
+				}
+				err = errors.Errorf("unexpected result length: %d", len(rows))
+			}
+		}
+		logutil.BgLogger().Warn("failed to get last insert id when adding runaway watch", zap.Error(err))
+		time.Sleep(time.Millisecond * time.Duration(retry*100))
```

Comment:
Better to use `select` to see `do.exit` is closed while sleeping.

Reply:
Got it, thx~